TACC/Core-CMS-Custom
# Task/fp 1816 extension requeste

## Overview

Addressing 500 errors on submission of form and fixing data types for database fields

## Related


- [FP-1816](https://jira.tacc.utexas.edu/browse/FP-1816)


## Changes

- Updated `ExtensionViewForm` to be a `TemplateView` to allow us to pass context data to the template
- Retrieving username from request now and passing in the username to the function to get submitter information
- Updated query to get the org name for a submitter as well 
- Added a session storage that stores a submitters information so that multiple database calls are not made to get submitter information
- Updated `applicable-data-period` value to be an int
- Updated `datetime` value to be str

## Testing

1. In the file `extension/views.py` replace line 24 with:

```
 submitters = [
        (2, 'TESTGOLD', 10000001, 'gmunoz1', 'CHCD'),
        (3, 'TESTGOLD', 10000002, 'gmunoz1', 'CHCD'),
        (1, 'TESTGOLD', 10000000, 'gmunoz1', 'CHCD')
    ]
```

2. Go to http://localhost:8000/submissions/extension-request/ and input values in each field 
3. Make sure form validation works as intended and that there is a selector at the bottom for business name
4. Dates can be in past or present. No limits on dates until UT Health says otherwise.

## UI

<img width="1915" alt="Screen Shot 2023-02-02 at 1 30 52 PM" src="https://user-images.githubusercontent.com/23081932/216431167-cd4f0501-9084-4800-ad5d-8f265792c3e5.png">


## Notes

- Once this is deployed to staging need to test if the database save works correctly and that there are no 500 errors
